File: backend/src/merchant/objects/Product.scala
package delivery.merchant.objects

import delivery.shared.objects.{InventoryStatus, ListingStatus, MerchantId, ProductId}

final case class ProductBundleOption(productId: ProductId, recommended: Boolean = false)

final case class ProductBundleGroup(
    id: String,
    name: String,
    quantity: Int,
    options: List[ProductBundleOption] = Nil
)

final case class Product(
    id: ProductId,
    merchantId: MerchantId,
    name: String,
    price: Double,
    description: String,
    imageUrl: String,
    monthlySales: Int,
    remainingStock: Int,
    listingStatus: ListingStatus,
    inventoryStatus: InventoryStatus,
    discountText: Option[String] = None,
    categoryName: String = "默认分类",
    bundleGroups: List[ProductBundleGroup] = Nil
)

File: frontend/src/objects/merchant/Product.ts
import type { InventoryStatus, ListingStatus, MerchantId, ProductId } from '@/objects/shared/ids'
import type { ProductBundleSelectionType } from './ProductBundleSelectionType'
import type { ProductInventoryMode } from './ProductInventoryMode'

export type { ProductInventoryMode } from './ProductInventoryMode'
export type ProductBundleGroupSelectionType = ProductBundleSelectionType

export interface ProductBundleOption {
  productId: ProductId
  recommended: boolean
  extraPrice: number
  customExtraPrice?: boolean
}

export interface ProductBundleGroup {
  id: string
  name: string
  quantity: number
  selectionType: ProductBundleGroupSelectionType
  includedPrice: number
  options: ProductBundleOption[]
}

export interface Product {
  id: ProductId
  merchantId: MerchantId
  name: string
  price: number
  description: string
  imageUrl: string
  categoryName?: string
  monthlySales: number
  remainingStock: number
  listingStatus: ListingStatus
  inventoryStatus: InventoryStatus
  inventoryMode?: ProductInventoryMode
  maxPerOrder?: number | null
  discountText?: string
  bundleGroups?: ProductBundleGroup[]
}
